src/pages/index.js

```diff
@@ -78,34 +111,103 @@ picturePopup.setEventListeners();
 const userInfo = new UserInfo({
   nameSelector: "#profile-title",
   jobSelector: "#profile-description",
+  avatarSelector: "#profile-image",
+});
+
+api.getCurrentUser().then((result) => {
+  userInfo.setUserInfo(result.name, result.about);
+  userInfo.setAvatar(result.avatar);
 });
 
 /******************
  * EVENT HANDLERS *
  ******************/
 function handleProfileEditSubmit(e, { title, description }) {
   e.preventDefault();
-  userInfo.setUserInfo(title, description);
-  profilePopup.close();
+  profilePopup.loading(true);
+  api
+    .updateUser(title, description)
+    .then((res) => res.json())
+    .then((result) => {
+      console.log(result);
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      userInfo.setUserInfo(title, description);
+      profilePopup.loading(false);
+      profilePopup.close();
+    });
 }
 
 function handleAddCardModalSubmit(e, cardData) {
   e.preventDefault();
-  cardSection.addItem(createNewCardEl(cardData));
-  addCardPopup.close();
+  addCardPopup.loading(true);
+  api
+    .createCard(cardData.name, cardData.link)
+    .then((res) => res.json())
+    .then((result) => {
+      cardData._id = result._id;
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      cardSection.addItem(createNewCardEl(cardData));
+      addCardPopup.loading(false);
+      addCardPopup.close();
+    });
+}
+
+function handleAvatarModalSubmit(e, { link }) {
+  e.preventDefault();
+  avatarModal.loading(true);
+  api
+    .updateAvatar(link)
+    .then((res) => res.json())
+    .then((result) => {
+      console.log(result);
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      userInfo.setAvatar(link);
+      avatarModal.loading(false);
+      avatarModal.close();
+    });
 }
 
 function handleCardClick(cardObject) {
   picturePopup.open(cardObject.link, cardObject.name);
 }
 
+function handleDeleteBtnClick(calledObj) {
+  deleteCardPopup.open(calledObj);
+}
+
+function handlePopupButtonClick(calledObj) {
+  calledObj.handleDelete();
+  api.deleteCard(calledObj.id);
+  deleteCardPopup.close();
+}
+
+function handleLikeBtnClick(calledObj) {
+  if (!calledObj.isLiked()) {
+    api.likeCard(calledObj.id);
+  } else {
+    api.dislikeCard(calledObj.id);
+  }
+  calledObj.setLikeBtnState();
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.